src/xdist/scheduler/customgroup.py

```diff
@@ -0,0 +1,349 @@
+    - 2 test cases are marked with @pytest.mark.high
+    - A pytest.ini file contains the following lines:
+[pytest]
+
+markers=
+    low: 4
+    medium: 2
+    high: 1
+
+Then the 4 low test cases will be ran on 4 workers (distributed evenly amongst the 4, as the load.py scheduler functions)
+Then the 4 medium test cases will be ran on 2 workers (again, distributed evenly), only after the low test cases are complete (or before they start).
+Then the 2 high test cases will be ran on 1 worker (distributed evenly), only after the low and medium test cases are complete (or before they start).
+
+This allows a pytest user more custom control over processing tests.
+One potential application would be measuring the resource utilization of all test cases. Test cases that are not
+resource intensive can be ran on many workers, and more resource intensive test cases can be ran once the low
+resource consuming tests are done on fewer workers, such that resource consumption does not exceed available resources.
+"""
+from __future__ import annotations
+
+from itertools import cycle
+from typing import Sequence
+
+import pytest
+
+from xdist.remote import Producer, WorkerInteractor
+from xdist.report import report_collection_diff
+from xdist.workermanage import parse_spec_config
+from xdist.workermanage import WorkerController
+
+class CustomGroup:
+    """
+    # TODO: update docs here
+    """
+
+    def __init__(self, config: pytest.Config, log: Producer | None = None) -> None:
+        self.terminal = config.pluginmanager.getplugin("terminalreporter")
+        self.numnodes = len(parse_spec_config(config))
+        self.node2collection: dict[WorkerController, list[str]] = {}
+        self.node2pending: dict[WorkerController, list[int]] = {}
+        self.pending: list[int] = []
+        self.collection: list[str] | None = None
+        if log is None:
+            self.log = Producer("loadsched")
+        else:
+            self.log = log.loadsched
+        self.config = config
+        self.maxschedchunk = self.config.getoption("maxschedchunk")
+        # TODO: Type annotation incorrect
+        self.dist_groups: dict[str, str] = {}
+        self.pending_groups: list[str] = []
+        self.is_first_time = True
+        self.do_resched = False
+
+    @property
+    def nodes(self) -> list[WorkerController]:
+        """A list of all nodes in the scheduler."""
+        return list(self.node2pending.keys())
+
+    @property
+    def collection_is_completed(self) -> bool:
+        """Boolean indication initial test collection is complete.
+
+        This is a boolean indicating all initial participating nodes
+        have finished collection.  The required number of initial
+        nodes is defined by ``.numnodes``.
+        """
+        return len(self.node2collection) >= self.numnodes
+
+    @property
+    def tests_finished(self) -> bool:
+        """Return True if all tests have been executed by the nodes."""
```

Comment:
Maybe we should call out the "more than one test remaining" concept.

Reply:
Just in the docstring?